### DISCUSSION TITLE
Various options when Compiling with Tasmocompiler and their size

### DISCUSSION THREAD

**Discussion by NsinghP** [benzino77/tasmocompiler#208] [Q&A]
Hello, 

Please look at questions 2 and 3 below:

When Compiling with newest Tasmocompiler @benzino77 ‘s site and using Wemos/NodeMCU option, we are presented with following options:

(1)Air/Gas Sensor (2) Amazon Alexa, (3) Arduino Slave, (4) Display Vcc, (5) Displays (12C/SPI) (6) Distance Sensors, (7) Domoticz, (8)Energy sensors, (9) Home Assistant, (10) IO port expander, (11) IR Support, (12) KNX, (13) Light sensors, (14) mDNS discovery, (15) MQTT over TLS, (16) RF transceiver, (17) Rules, (18) Script, (19) SD card/LittleFS, (20) Temp/Hum sensor, (21) Timers, (22) Tuya MCU, (23) web interface, (24) WS2812 Leds, (25) Zigbee.
Out of these 4 are already selected by default as follows:
(17) Rules, (19) SD card/LittleFS, (21) Timers, (23) Web Interface

In addition to above four already selected by default I am using two more as follows:
(7) Domoticz, (15) MQTT over TLS
Bringing the options selected by me to a total of six. Resulting .bin file is working good and there are no problems when I upgrade older versions of tasmota with this new .bin file using firmware.bin.gz version which is 452 KB in size.

My questions are:

1) Can I keep adding more options above and beyond the six I am already using without any problem. What is the max size that .bin.gz file can be without running into problems flashing/upgrading. I am using ESP8266 12F based device with 4MB flash. (This question 1 has already been answered by ascillato  Please see https://github.com/arendst/Tasmota/discussions/12379 )

2) It would be nice to read up on what all these option do. Is there a link/site which covers explanation for all the options in one place?

3) It would also be nice to see right after the option name how much size it is so it is easy to calculate when you have reached max size for your .bin file. Can this be done?

Regards,

Navi

**Comment by benzino77** [comment]
Have a look my answer [here](https://github.com/arendst/Tasmota/discussions/12379#discussioncomment-877722)

**Reply by NsinghP** [reply]
> 
> 
> Have a look my answer [here](https://github.com/arendst/Tasmota/discussions/12379#discussioncomment-877722)

Thanks
